src/Widgets/TakeButton.vala

```diff
@@ -0,0 +1,112 @@
+ /*
+ * Copyright (c) 2011-2016 elementary LLC. (https://github.com/elementary/camera)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ * Authored by: Alain M. <alain23@protonmail.com>
+ */
+
+public class Camera.Widgets.TakeButton : Gtk.Button {
+    private Gtk.Label timer_label;
+    private Gtk.Image take_image;
+
+    private bool timer_active = false;
+
+    public const string TAKE_BUTTON_STYLESHEET = """
+        .take-button {
+            border-radius: 400px;
+        }
+    """;
+
+    public string icon_name {
+        set { take_image.icon_name = value; }
+    }
+
+    public TakeButton () {
+        Object (
+            sensitive: false,
+            width_request: 54
+        );
+     }
+
+     construct {
+         var main_grid = new Gtk.Grid ();
```

Comment:
Probably don't need to declare this early. You could group it down on line 69 so that everything is together